### PR TITLE
Adding new teams default firstprivate test

### PR DESCRIPTION
**Pull request by jrreap**
New test to confirm if the teams construct properly handles default(firstprivate) when working with variables in that region.

### CODE REVIEW

File: tests/5.1/teams/test_teams_default_firstprivate.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+int main() {
+	int a[N];
+	int errors = 0;
+	int not_shared = 5;
+
+	for (int i = 0; i < N; i++) {
+		a[i] = i;
+	}
+
+	#pragma omp teams default(firstprivate) num_teams(OMPVV_NUM_TEAMS_DEVICE)
```
Comment (tmh97) [inline tests/5.1/teams/test_teams_default_firstprivate.c:32]: Wrap {} around everything you wish to be included in the teams region 
 	#pragma omp teams default(firstprivate) num_teams(OMPVV_NUM_TEAMS_DEVICE) {
                    code that goes in
                    your structured block
         }

File: tests/5.1/teams/test_target_teams_default_firstprivate.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	int a[N];
+	int errors = 0;
+	int not_shared = 5;
+
+	for (int i = 0; i < N; i++) {
+		a[i] = i;
+	}
+
+	#pragma omp target
+	#pragma omp teams default(firstprivate) num_teams(OMPVV_NUM_TEAMS_DEVICE)
```
Comment (tmh97) [inline tests/5.1/teams/test_target_teams_default_firstprivate.c:33]: This works, but try a combined construct! `#pragma omp target teams default ( etc ....) etc ...`

File: tests/5.1/teams/test_teams_default_firstprivate.c (hunk 1 above)
Comment (tmh97) [reply tests/5.1/teams/test_teams_default_firstprivate.c:32]: https://github.com/SOLLVE/sollve_vv/blob/master/tests/5.0/teams/test_teams.c -> leaving this here for meeting, will chat about it then!

**Review by tmh97 — APPROVED** [review]

**Comment by tmh97** [comment]
This is good-to-go :) @spophale @nolanbaker31

File: tests/5.1/teams/test_target_teams_default_firstprivate.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#define N 1024
+
+int main() {
+	int errors = 0;
+	int not_shared = 5;
+	int num_teams = 0; 
+
+	#pragma omp target teams default(firstprivate) map(tofrom:num_teams) shared(num_teams) num_teams(OMPVV_NUM_TEAMS_DEVICE)
+	{
+		num_teams = omp_get_num_teams();
```
Comment (tmh97) [inline tests/5.1/teams/test_target_teams_default_firstprivate.c:30]: Race condition on omp_get_num_teams(), wrap line 30 in an if-statement that checks if omp_get_team_num() == 0

File: tests/5.1/teams/test_target_teams_default_firstprivate.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int main() {
+	int errors = 0;
```
Comment (tmh97) [inline tests/5.1/teams/test_target_teams_default_firstprivate.c:24]: Add -> `OMPVV_TEST_OFFLOADING;` here

**Review by nolanbaker31 — APPROVED** [review]
looks good to go!

**Pull request merged**